infra/bridge/src/controllers/hooks/index.js

```diff
@@ -0,0 +1,184 @@
+      redirectUrl: '/hooks/twitter/__auth-redirect'
+    })
+
+    oAuthToken = twitterResponse.oAuthToken
+    oAuthTokenSecret = twitterResponse.oAuthTokenSecret
+  } catch (error) {
+    logger.error(error)
+    return res.status(500).send({
+      errors: ['Failed to get Twitter OAuth request token.']
+    })
+  }
+
+  req.session.oAuthToken = oAuthToken
+  req.session.oAuthTokenSecret = oAuthTokenSecret
+
+  const url =
+    constants.TWITTER_BASE_AUTH_URL +
+    querystring.stringify({ oauth_token: oAuthToken })
+
+  return res.redirect(url)
+})
+
+/**
+ * Twitter OAuth callback
+ * Creates and susbcribes to webhook if everything goes well
+ */
+router.get('/twitter/__auth-redirect', async (req, res) => {
+  const session = await req.sessionStore.get(req.query.sid)
+
+  if (!session || !session.oAuthToken || !session.oAuthTokenSecret) {
+    return res.status(400).send({
+      errors: ['Invalid Twitter oAuth session.']
+    })
+  }
+
+  let oAuthAccessToken, oAuthAccessTokenSecret
+  try {
+    const accessToken = await getTwitterOAuthAccessToken(
+      session.oAuthToken,
+      session.oAuthTokenSecret,
+      req.query.oauth_verifier
+    )
+    oAuthAccessToken = accessToken.oAuthAccessToken
+    oAuthAccessTokenSecret = accessToken.oAuthAccessTokenSecret
+  } catch (error) {
+    if (error.statusCode == 401) {
+      return res.status(401).send({
+        errors: ['The oauth-verifier provided is invalid.']
+      })
+    }
+    logger.error(error)
+    return res.status(500).send({
+      errors: ['Could not get a Twitter access token.']
+    })
+  }
+
+  let userProfileData
+  try {
+    userProfileData = await verifyTwitterCredentials(
+      oAuthAccessToken,
+      oAuthAccessTokenSecret
+    )
+  } catch (error) {
+    logger.error(error)
+    return res.status(500).send({
+      errors: ['Could not verify Twitter credentials.']
+    })
+  }
+
+  if (
+    userProfileData.username.toLowerCase() !==
+    process.env.TWITTER_ORIGINPROTOCOL_USERNAME.toLocaleLowerCase()
+  ) {
+    return res.status(400).send({
+      errors: ['Invalid account']
+    })
+  }
+
+  try {
+    await subscribeToHooks(oAuthAccessToken, oAuthAccessTokenSecret)
+  } catch (err) {
+    logger.error(err)
+    return res.status(400).send({
+      success: false,
+      errors: [`Failed to subscribe: ${err.message}`]
+    })
+  }
+
+  return res.status(200).send({
+    success: true
+  })
+})
+
+/**
+ * Webhook Authorization
+ */
+router.get('/twitter', (req, res) => {
+  const hmac = crypto
+    .createHmac('sha256', process.env.TWITTER_ORIGIN_CONSUMER_SECRET)
+    .update(req.query.crc_token)
+    .digest('base64')
+
+  res.status(200).send({
+    response_token: `sha256=${hmac}`
+  })
+})
+
+/**
+ * Twitter posts events to this endpoint
+ * Should always return 200 with no response
+ */
+router.post('/twitter', (req, res) => {
+  if (req.body.follow_events) {
+    // Follow event(s)
+    const events = req.body.follow_events
+    events.forEach(event => {
```

Comment:
Note: we could use a Promise.all to parallelize the request to redis rather than handling them in a serial way. But if we do that we should make sure to not go over the max number of concurrent redis connections and may need chunking. So perhaps just add a TODO. If we start noticing the webhook takes a long time, we can optimize. We'll get very tiny volume initially so it will be fast enough how you have it currently.

Same comment for forEach statement on line 170.